tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/OnBoardMember.kt

```
@@ -187,21 +188,48 @@ class OnBoardMember : Runnable, BaseOnboard() {
         assignSoftHsmAndGenerateKey("LEDGER")
     }
 
-    override val registrationContext by lazy {
-        val preAuth = if (preAuthToken != null) mapOf("corda.auth.token" to preAuthToken) else emptyMap()
-        val endpoints = mutableMapOf<String, String>()
+    private val notaryKeyId by lazy {
+        assignSoftHsmAndGenerateKey("NOTARY")
+    }
 
-        p2pGatewayUrls.mapIndexed { index, url ->
-            endpoints["corda.endpoints.$index.connectionURL"] = url
-            endpoints["corda.endpoints.$index.protocolVersion"] = "1"
+    override val registrationContext by lazy {
+        val preAuth = preAuthToken?.let { mapOf("corda.auth.token" to it) } ?: emptyMap()
+        val roleProperty = if (role == "notary") mapOf("corda.roles.0" to role) else emptyMap()
+        val notaryServiceName = customProperties.firstNotNullOfOrNull { property ->
+            val key = property.substringBefore("=")
+            val value = property.substringAfter("=")
+            if (key == "corda.notary.service.name") value else null
+        }
+        val notaryProperties = if (role == "notary") {
+            mapOf(
+                "corda.notary.service.name" to notaryServiceName,
```

If we are a notary role and the `notaryServiceName,` is null, should we throw an exception?

---

We can do something like this `val notaryProperties = if (role == "notary" && notaryServiceName != null)`

---

I don't think we support a notary without a service name.